hollow/src/main/java/com/netflix/hollow/api/consumer/data/AbstractHollowDataAccessor.java

```diff
@@ -81,6 +82,17 @@ public AbstractHollowDataAccessor(HollowReadStateEngine rStateEngine, String typ
         }
     }
 
+    /**
+     * Indicate whether Data Accessor contains prior state
+     *
+     * NOTE: This is critical since loading a Snapshot will not contain any information about changes from prior state
+     *
+     * @return true indicate it contains prior state
+     */
+    public boolean hasPriorState() {
```

Comment:
Can you plz share any usage of this method? It'll help me better understand why its needed

Reply:
On a snapshot, getAddedRecords() will return all Ids so this is the key indicator to enable properly interpret that the added ids were not from delta from prior state.